dist/ember-charts.js

```diff
@@ -1381,9 +1381,17 @@ define('ember-charts/components/pie-chart', ['exports', 'module', 'ember', './ch
       groups.select('path').attr(this.get('sliceAttrs'));
 
       var labelWidth = this.get('labelWidth');
+      var maxLabelWidth = this.get('outerWidth') / 2 - this.get('labelPadding');
       var labelTrimmer = _LabelTrimmer['default'].create({
-        getLabelSize: function getLabelSize() {
-          return labelWidth;
+        getLabelSize: function getLabelSize(d, selection) {
+          var text, xPos;
+          // Pick the current text element by filtering on data
+          text = selection.filter(function (data, i) {
+            if (data === d) return this;
+          });
```

Comment:
Also, this should actually just be `return data === d`, because `filter` expects a boolean.